Corrected error and warnings for bulk statistics when feature labels do not match labels in segmentation mask

JuliaKukulies opened the pull request:
Solves #446 


* [x] Have you followed our guidelines in CONTRIBUTING.md? 
* [x] Have you self-reviewed your code and corrected any misspellings? 
* [x] Have you written documentation that is easy to understand?
* [x] Have you written descriptive commit messages? 
* [x] Have you added NumPy docstrings for newly added functions? 
* [x] Have you formatted your code using black? 
* [x] If you have introduced a new functionality, have you added adequate unit tests?
* [x] Have all tests passed in your local clone? 
* [x] If you have introduced a new functionality, have you added an example notebook?
* [x] Have you kept your pull request small and limited so that it is easy to review? 
* [x] Have the newest changes from this branch been merged?

JuliaKukulies commented:
Tests should be OK after merge of #451

w-k-jones commented on tobac/utils/bulk_statistics.py:
```diff
@@ -301,16 +307,14 @@ def get_statistics_from_mask(
-                "The labels of the segmentation mask and the feature dataframe do not seem to match. Please make sure you provide the correct input feature dataframe to calculate the bulk statistics. "
-            )
+        # continue loop because not all timesteps might have matching IDs
+        if not np.any(np.isin(features_t[id_column], np.unique(segmentation_mask_t))):
+            warnings.warn("Not all timesteps have matching features", UserWarning)
+            continue
```
This may results in timesteps without matching features/segments missing from the output dataframe. We should append the input features as is and hope pd.concat sorts it out.

JuliaKukulies replied:
addressed this in latest commit :)

JuliaKukulies commented:
this now also addressed #447, adds some tolerance to the timestep selection in the bulk statistic calculation, fixed the issue related to appended timesteps when there are no matching labels and added a test to ensure that all timesteps of feature input dataframe are also found in output dtaframe after the statistics calculation. Ready to review @freemansw1 and if you wanna have a second look @w-k-jones

freemansw1 reviewed (approved):
Thanks, @JuliaKukulies . No concerns here.

freemansw1 commented:
> Thanks, @JuliaKukulies . No concerns here.

Actually, I missed that the notebook build is failing. Can you check on that? The other two seem to be unrelated issues.

JuliaKukulies commented:
Good catch, I corrected to typos which I think ended up there because i used the notebook for some testing on that branch. Thanks for the review @freemansw1

w-k-jones commented:
The Zenodo formatting issue seems to be an issue with the action failing to install python, I'm happy for you to merge and we can fix the actions at a later date.
